processes/dags/pubmed_migration.py

```diff
@@ -111,59 +105,247 @@ def make_tarfile(output_filepath: str, source_dir: str) -> None:
+
+    Args:
+        xml_path (str): path to input XML file
+        json_path (str): path to destination JSON file
+    """
+    pubmed_xml = pp.parse_pubmed_xml(xml_path)
+    pubmed_paragraph = pp.parse_pubmed_paragraph(xml_path)
+    pubmed_references = pp.parse_pubmed_references(xml_path)
+
+    text = " ".join([p["text"] for p in pubmed_paragraph])
+    case_report = {
+        "pmID": pubmed_xml["pmid"],
+        "messages": [],
+        "source_files": [],
+        "modifications": [],
+        "normalizations": [],
+        # ctime            : 1351154734.5055847,
+        "text": text,
+        "entities": [],
+        "attributes": [],
+        # date : { type: Date, default: Date.now }
+        "relations": [],
+        "triggers": [],
+        "events": [],
+        "equivs": [],
+        "comments": [],
+        # sentence_offsets     : [],
+        # token_offsets    : [],
+        "action": None,
+        "abstract": pubmed_xml["abstract"],
+        "authors": pubmed_xml["author_list"],
+        "keywords": [],
+        "introduction": None,
+        "discussion": None,
+        "references": []
+    }
+
+    return case_report
+
 def extract_pubmed_data() -> None:
     """Extracts case-reports from pubmed data and stores result on S3
     """
-    logging.info("start import wrapper")
-    username = "anonymous"
-    password = "ifso6888@gmail.com"
-    ftp_url = 'ftp.ncbi.nlm.nih.gov'
-    ftp_path = 'pub/pmc/oa_bulk'
+
+    # to test specific tar files
+    pattern = 'non_comm_use.A-B.xml.tar.gz'
+    #pattern = "*.xml.tar.gz"
+    ftp_path = '/pub/pmc/oa_bulk'
     root_dir = '/usr/local/airflow'
-    temp_dir = root_dir + '/' + 'temp'
+    temp_dir = os.path.join(root_dir, 'temp')
+    bucket_name = 'supreme-acrobat-data'
+    prefix = 'case_reports/pubmed/original'
 
-    ftp = FTP(ftp_url)
-    ftp.login(username, password)
-    ftp.cwd(ftp_path)
-    filenames = ftp.nlst('*.xml.tar.gz')
-    try:
-        ftp.quit()
-    except:
-        None
+    filenames = ftp_hook.list_directory(ftp_path)
+    filenames = list(
+        filter(lambda filename: fnmatch.fnmatch(filename, pattern), filenames))
 
     for filename in filenames:
-        ftp = FTP(ftp_url)
-        ftp.login(username, password)
-        ftp.cwd(ftp_path)
+        create_dir(temp_dir)
+        remote_path = os.path.join(ftp_path, filename)
+        local_path = os.path.join(temp_dir, filename)
+
+        ftp_hook.retrieve_file(remote_path, local_path)
+        o_path = extract_original_name(local_path)
+
+        extract_file_case_reports(local_path, o_path)
+        delete_file(local_path)
+        make_tarfile(local_path, o_path)
+        key = os.path.join(prefix, filename)
+        s3_hook.load_file(
+            local_path, key, bucket_name=bucket_name, replace=True)
+        delete_dir(temp_dir)
+
 
+def extract_pubmed_data_failure_callback(context) -> None:
+    delete_temp()
+
+
+def transform_pubmed_data() -> None:
```

Comment:
Let's try to refactor this. Nothing wrong per se, but the function is getting kinda big and hard to follow. We can remove unnecessary code, and maybe create functions for chunks of code that are complex.